pkg/kubelet/dockertools/manager.go

```diff
@@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
 					break
 				}
 			}
+			if pod.Spec.TerminationGracePeriodSeconds != nil {
+				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
```

Comment:
From v1beta3/types.go:
```
	// DeletionTimestamp is the time after which this resource will be deleted. This
	// field is set by the server when a graceful deletion is requested by the user, and is not
	// directly settable by a client. The resource will be deleted (no longer visible from
	// resource lists, and not reachable by name) after the time in this field. Once set, this
	// value may not be unset or be set further into the future, although it may be shortened
	// or the resource may be deleted prior to this time. For example, a user may request that
	// a pod is deleted in 30 seconds. The Kubelet will react by sending a graceful termination
	// signal to the containers in the pod. Once the resource is deleted in the API, the Kubelet
	// will send a hard termination signal to the container.
	DeletionTimestamp *util.Time
...
	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
	// specified type will be used.
	GracePeriodSeconds *int64
...
	// Optional duration in seconds the pod needs to terminate gracefully. May be decreased in delete request.
	// Value must be non-negative integer. The value zero indicates delete immediately.
	// If this value is nil, the default grace period will be used instead.
	// The grace period is the duration in seconds after the processes running in the pod are sent
	// a termination signal and the time when the processes are forcibly halted with a kill signal.
	// Set this value longer than the expected cleanup time for your process.
	TerminationGracePeriodSeconds *int64
```

TerminationGracePeriodSeconds is intended to be the default grace period, such as that used by replication controller and node controller, or by users in the case that they don't override it. DeletionTimestamp is intended to reflect the actual grace period. It should be initialized from GracePeriodSeconds. If no value is provided for GracePeriodSeconds, it should be initialized from TerminationGracePeriodSeconds (for pods).